### PR TITLE
test/e2e: more comprehensive test for NodeFeature objects

### PR DESCRIPTION
**Pull request by marquiz**
Test creation of multiple NodeFeature objects per node, mocking 3rd party extensions.

### CODE REVIEW

**Review by fmuyassarov — COMMENTED** [review]
Left some comments, but otherwise it is really good to have more tests indeed for the NodeFeature.

File: test/e2e/data/nodefeature-2.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: nfd.k8s-sigs.io/v1alpha1
+kind: NodeFeature
+metadata:
+  name: zzz-e2e-features-2
+spec:
+  features:
+    flags: {}
+    attributes: {}
+    instances: {}
```
[the remark this answers is not in the document]
Comment (marquiz) [reply test/e2e/data/nodefeature-2.yaml]: Yeah, makes sense, now. Deleted

File: test/e2e/utils/crd.go [hunk 2]
```diff
@@ -59,6 +59,37 @@ func CreateNfdCRDs(cli extclient.Interface) ([]*apiextensionsv1.CustomResourceDe
...
+		obj.Labels[nfdv1alpha1.NodeFeatureObjNodeNameLabel] = nodename
+
+		if _, err := cli.NfdV1alpha1().NodeFeatures(namespace).Get(context.TODO(), obj.Name, metav1.GetOptions{}); errors.IsNotFound(err) {
+			if _, err := cli.NfdV1alpha1().NodeFeatures(namespace).Create(context.TODO(), obj, metav1.CreateOptions{}); err != nil {
+				return names, fmt.Errorf("failed to create NodeFeature %w", err)
+			}
+		} else if err == nil {
+			if _, err = cli.NfdV1alpha1().NodeFeatures(namespace).Update(context.TODO(), obj, metav1.UpdateOptions{}); err != nil {
+				return names, fmt.Errorf("failed to update NodeFeature object: %w", err)
+			}
```
[the remark this answers is not in the document]
Comment (marquiz) [reply test/e2e/utils/crd.go:85]: Good catch 👏 Changed as suggested.

File: test/e2e/node_feature_discovery.go [hunk 3]
```diff
@@ -494,6 +494,94 @@ var _ = SIGDescribe("Node Feature Discovery", func() {
...
+
+					By("Waiting for worker daemonset pods to be ready")
+					Expect(testpod.WaitForReady(f.ClientSet, f.Namespace.Name, workerDS.Spec.Template.Labels["name"], 5)).NotTo(HaveOccurred())
+
+					By("Verifying node labels from nfd-worker")
+					expectedLabels = map[string]k8sLabels{
+						"*": {
+							"feature.node.kubernetes.io/fake-fakefeature1": "true",
+							"feature.node.kubernetes.io/fake-fakefeature2": "true",
+							"feature.node.kubernetes.io/fake-fakefeature3": "true",
```
[the remark this answers is not in the document]
Comment (marquiz) [reply test/e2e/node_feature_discovery.go]: Agree, changed as suggested

**Comment by marquiz** [comment]
@fmuyassarov thanks for the review. Comments addressed. I rebased the PR on top of the latest master so that the tests are runnable again (accommodating for the NodeFeature fields becoming optional and being removed from the test data yaml). PTAL

**Comment by fmuyassarov** [comment]
Thanks @marquiz , looks good now.
/lgtm

**Pull request merged**